src/dsp_tools/utils/xml_upload.py

```diff
@@ -244,8 +251,8 @@ def _stash_circular_references(
                                 stashed_resptr_props[res][link_prop].append(str(value.value))
                         link_prop.values.remove(value)
             else:
-                raise BaseError(f'ERROR in remove_circular_references(): link_prop.valtype is '
-                                f'neither text nor resptr.')
+                logger.exception("ERROR in remove_circular_references(): link_prop.valtype is neither text nor resptr.")
+                raise BaseError("ERROR in remove_circular_references(): link_prop.valtype is neither text nor resptr.")
```

Comment:
is it necessary to log this here? since you raise the exception, you should be able to log it when you catch it again (and there's no point in logging the same exception twice)

Reply:
I quickly checked it, and it seems that raising an Error with `raise BaseError()` does NOT automatically get logged. I explicitly have to log the error before raising it. So I don't think there will be a doubling of the stack traces.

Or do you think that this architecture (log error, then raise it) is suboptimal?